### DISCUSSION TITLE
Docker compose + nginx - how to generate correct signature?

### DISCUSSION THREAD

**Discussion by mateuszwojt** [minio/minio#14363] [Q&A] [answered]
I'm using the default setup for running minio with nginx proxy in docker-compose, as described [here](https://docs.min.io/docs/deploy-minio-on-docker-compose.html). I'm also running a simple Express server inside the same docker-compose, which is exposing the endpoint for generating pre-signed URLs. I'm initializing my client like this:

```
let minioClient = new minio.Client({
  endPoint: "nginx",  // name of the container running the proxy
  port: 9000,
  useSSL: false,
  accessKey: ...,
  secretKey: ...,
});
```

To test the endpoint, I'm calling it in Postman (using localhost, as I don't have any DNS to redirect to containers by name). I'm getting back `S3Error: The request signature we calculated does not match the signature you provided. Check your key and signing method.`. 

If my endpoint looks like this: `http://localhost:4000/upload?objectName=someobject.jpg`, does it mean that the Host header contains `localhost` instead of `nginx` and that's why the signature calculation fails? How can I forward the headers correctly when I'm working with docker-compose?

**Comment by harshavardhana** [comment, marked as answer]
> If my endpoint looks like this: `http://localhost:4000/upload?objectName=someobject.jpg`, does it mean that the Host header contains `localhost` instead of `nginx` and that's why the signature calculation fails? How can I forward the headers correctly when I'm working with docker-compose?

Correct - you can't change hostnames on your own.
